File: backend/Services/Parsers/TradleParser.cs
using System.Text.RegularExpressions;

namespace backend.Services.Parsers
{
    public class TradleParser : BasicScoreResultParser
    {
        private ILogger<TradleParser> _logger;

        public TradleParser(ILogger<TradleParser> logger) : base(logger)
        {
            _logger = logger;
        }

        public override bool CountWinner => true;
        public override bool Default => false;
        public override string GameName => "Tradle";
        public override string? HelpText => null;
        protected override Regex Parser => new Regex($"#{GameName} #\\d+ (?<{ScoreGroup}>[\\d|X])/\\d");
        protected override string? ExtraContent => Url;
        public override string Url => "https://oec.world/en/tradle";
    }
}

File: lambdas/src/parsers/tradle-parser.ts
import { BasicScoreResultParser } from './base/basic-score-parser';

export class TradleParser extends BasicScoreResultParser {
  readonly countWinner = true;
  readonly gameName = 'Tradle';
  readonly helpText = null;
  readonly url = 'https://oec.world/en/games/tradle';
  protected readonly extraContent = this.url;
  
  // Note: C# uses [\d|X] which incorrectly includes literal pipe - fixed here to [\dX]
  protected readonly parser = new RegExp(
    `#${this.gameName} #\\d+ (?<${BasicScoreResultParser.SCORE_GROUP}>[\\dX])/\\d`
  );
}
